### DISCUSSION TITLE
Implement for Online Statue in WebV2

### DISCUSSION THREAD

**Discussion by IamTaoChen** [lejianwen/rustdesk-api#59] [General]
讨论是否需要完善设备的在线检测（web中）。如果要完善的话，从api服务器还是从ID 服务器去实现比较好。

**Comment by IamTaoChen** [comment]
ID server上已经完善，见 https://github.com/IamTaoChen/rustdesk-server/commit/f14ab4e73daf7acbb657b55ebd32deeb6d39dcf7

docker image:  [iamtaochen/rustdesk-server:latest](https://hub.docker.com/r/iamtaochen/rustdesk-server)
<img width="1417" alt="image" src="https://github.com/user-attachments/assets/44e7b9aa-86dc-4cdd-b7a4-1660425e5470">

**Reply by lejianwen** [reply]
牛逼，666

**Reply by zhangbo8418** [reply]
> ID server上已经完善，见 [IamTaoChen/rustdesk-server@f14ab4e](https://github.com/IamTaoChen/rustdesk-server/commit/f14ab4e73daf7acbb657b55ebd32deeb6d39dcf7)
> 
> docker image: [iamtaochen/rustdesk-server:latest](https://hub.docker.com/r/iamtaochen/rustdesk-server) <img alt="image" width="1417" src="https://private-user-images.githubusercontent.com/42793494/387583583-44e7b9aa-86dc-4cdd-b7a4-1660425e5470.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwMTY5ODQsIm5iZiI6MTczMjAxNjY4NCwicGF0aCI6Ii80Mjc5MzQ5NC8zODc1ODM1ODMtNDRlN2I5YWEtODZkYy00Y2RkLWI3YTQtMTY2MDQyNWU1NDcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE5VDExNDQ0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlNGQ0ODdkYWY0OTdhNmY4ZTUzNjJlNThlOGVjYjNiNGM0MDU4MzI1OGEzNzliOWU2OTE3OGU4MTg0OTk0OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gmYq_NQElizEwWzOi-tqZUAu8q63QPIqsboHDE53f3E">

NB是NB，但是地址簿还是不对。。。

**Reply by IamTaoChen** [reply]
> > ID server上已经完善，见 [IamTaoChen/rustdesk-server@f14ab4e](https://github.com/IamTaoChen/rustdesk-server/commit/f14ab4e73daf7acbb657b55ebd32deeb6d39dcf7)
> > docker image: [iamtaochen/rustdesk-server:latest](https://hub.docker.com/r/iamtaochen/rustdesk-server) <img alt="image" width="1417" src="https://private-user-images.githubusercontent.com/42793494/387583583-44e7b9aa-86dc-4cdd-b7a4-1660425e5470.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwMTY5ODQsIm5iZiI6MTczMjAxNjY4NCwicGF0aCI6Ii80Mjc5MzQ5NC8zODc1ODM1ODMtNDRlN2I5YWEtODZkYy00Y2RkLWI3YTQtMTY2MDQyNWU1NDcwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE5VDExNDQ0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlNGQ0ODdkYWY0OTdhNmY4ZTUzNjJlNThlOGVjYjNiNGM0MDU4MzI1OGEzNzliOWU2OTE3OGU4MTg0OTk0OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gmYq_NQElizEwWzOi-tqZUAu8q63QPIqsboHDE53f3E">
> 
> NB是NB，但是地址簿还是不对。。。

<img width="1320" alt="image" src="https://github.com/user-attachments/assets/25ba1ffa-bdf7-4afd-bbe6-40d0b3a03fad">
我这边显示没问题

**Reply by zhangbo8418** [reply]
因为它只显示第一个设备，后面的设备状态都不对

![Mac](https://github.com/user-attachments/assets/d080ceca-7cfd-4277-93a3-506093bf54d2)
![TV](https://github.com/user-attachments/assets/1ba06e7a-549d-4704-b511-318c339e940d)

**Comment by IamTaoChen** [comment]
问题出现在js代码中 
```js
(s[c / 8] & C) === C ? l.push(e[c]) : E.push(e[c])
```
`c / 8` 会得到小数

**Comment by IamTaoChen** [comment]
@lejianwen 你能修改一下js代码吗？我的本地测试服务器这两天下线了...
```js
(s[Math.floor(c / 8) ] & C) === C ? l.push(e[c]) : E.push(e[c])
```

**Reply by zhangbo8418** [reply]
我试了，好像可以了。

**Reply by lejianwen** [reply]
> @lejianwen 你能修改一下js代码吗？我的本地测试服务器这两天下线了...
> 
> ```js
> (s[Math.floor(c / 8) ] & C) === C ? l.push(e[c]) : E.push(e[c])
> ```

好的，我修改一下

**Reply by zhangbo8418** [reply]
web已经1.3.4了，是不是更新一下试试？现在的版本移动设备浏览器控制电脑木有鼠标指针

**Reply by lejianwen** [reply]
好，我下午更新下

**Reply by zhangbo8418** [reply]
移动端浏览器控制PC还是木有鼠标指针，感觉更了个寂寞，是不是它还有别的玩意儿啊？

**Reply by zhangbo8418** [reply]
哦，好像多了个文件传输
